core/Piwik.php

```diff
@@ -262,6 +263,51 @@ public static function checkUserHasSuperUserAccessOrIsTheUser($theUser)
         }
     }
 
+    /**
+     * Request a token auth to authenticate in a request.
+     *
+     * Note: During one request the token is only being requested once and used throughout the request. So you want to make
+     * sure the token is valid for enough time for the whole request to finish.
+     *
+     * @param string $reason some short string/text explaining the reason for the token generation, eg "CliMultiAsyncHttpArchiving"
+     * @param int $validForHours For how many hours the token should be valid. Should not be valid for more than 14 days.
+     * @return mixed
+     */
+    public static function requestTemporarySystemAuthToken($reason, $validForHours)
+    {
+        static $token = array();
```

Comment:
Would this ever need to be reset for tests? If so, might be good to keep it as a class static variable.

Reply:
I don't think it's needed for tests. If so, we can refactor it for sure 👍